### ISSUE TITLE
LRS2分离效果

### ISSUE THREAD

**Issue by shengzhang0222** [JusperLee/TDANet#9]
你好，目前尝试在LRS2数据集上复现论文效果，参数配置如下：

### Network config
audionet:
  audionet_name: TDANet
  audionet_config:
    out_channels: 128
    in_channels: 512
    num_blocks: 16
    upsampling_depth: 5
    enc_kernel_size: 4 # ms
    num_sources: 2

### Data config
datamodule:
  data_name: LRS2DataModule
  data_config:
    train_dir: DataPreProcess/lrs2_json/tr
    valid_dir: DataPreProcess/lrs2_json/cv
    test_dir: DataPreProcess/lrs2_json/tt
    n_src: 2
    sample_rate: 16000
    segment: 2.0
    normalize_audio: false
    batch_size: 32
    num_workers: 8
    pin_memory: true
    persistent_workers: false

目前训练出来的效果：
Monitored metric val_loss/dataloader_idx_0 did not improve in the last 30 records. Best score: -10.659. Signaling Trainer to stop.
Epoch 445
loss: -12.3 v_num: 1 train_loss_step: -11.964 val_loss/dataloader_idx_0: -10.651  test_loss/dataloader_idx_1: -10.879 lr: 0.0 train_loss_epoch: -12.301

使用的模型为TEDNet.py，看起来和论文里 13.2的SI-SNRi差距很大，请问能帮忙看下吗？谢谢

**Comment by JusperLee** [comment]
1. 你可以尝试使用TDANetBest这个模型来训练。
2. 另一个是你现在的batch_size是我之前的8倍，可能需要调整learning rate来匹配batch增大导致的问题。

**Comment by shengzhang0222** [comment]
感谢你的建议！我重新尝试下。

**Comment by shengzhang0222** [comment]
你好，再请问下，我的实验和论文的差异，就这两个地方吗？
1、模型：TDANet  -> TDANetBest
2、batchsize ：32 -> 4

希望能先完全对齐配置和模型，复现论文的结果，谢谢

**Comment by JusperLee** [comment]
> 你好，再请问下，我的实验和论文的差异，就这两个地方吗？ 1、模型：TDANet -> TDANetBest 2、batchsize ：32（单卡） -> 4
> 
> 希望能先完全对齐配置和模型，复现论文的结果，谢谢

是的。我今天还会上传预训练模型。

**Comment by JusperLee** [comment]
During the process of uploading the pre-trained models, I discovered some issues with the TDANetBest model. I've since updated the model files.

In addition, I've provided two pre-trained models trained on the LRS2 dataset. You can find them on the Hugging Face Model Hub at the following URLs:

1. [TDANetBest-2ms-LRS2](https://huggingface.co/JusperLee/TDANetBest-2ms-LRS2)
2. [TDANetBest-4ms-LRS2](https://huggingface.co/JusperLee/TDANetBest-4ms-LRS2)

To utilize these models, you can use the following code:

```python
import os
import torch
import look2hear.models

os.environ['CUDA_VISIBLE_DEVICES'] = "0"

model = look2hear.models.BaseModel.from_pretrain("JusperLee/TDANetBest-2ms-LRS2").cuda()
test_data = torch.randn(1, 1, 16000).cuda()
out = model(test_data)
print(out.shape)
```

Please let me know if you encounter any issues or if there's anything else I can assist you with

**Comment by shengzhang0222** [comment]
感谢分享预训练模型！
我尝试使用TDANetBest，并且修改了学习率，看起来提升比较明显，后续如果有问题再请教，谢谢。

**Comment by chiexplorer** [comment]
> 感谢分享预训练模型！ 我尝试使用TDANetBest，并且修改了学习率，看起来提升比较明显，后续如果有问题再请教，谢谢。

请问您后来是如何调整学习率的，是保持batch size=32然后增大了学习率吗

**Comment by JusperLee** [comment]
一般来说batch size和lr成倍数增加

**Comment by chiexplorer** [comment]
暂时不在线，有事情留言

**Comment by chiexplorer** [comment]
感谢你的解答，我在进行LRS2-2mix数据集性能复现的过程中遇到了一个困惑的地方，是关于TDANetBest在每个epoch上的训练耗时。
问题描述：
	我注意到TDANetBest在单个epoch上的训练时间约为35分钟。当我检查GPU和CPU的利用率时，我观察到GPU利用率在30%上下波动，CPU利用率约为120%，内存占用约2G，GPU显存占用约15G。
	然后，我保持超参数不变，在TDANet模型上进行类似的测试，结果为：单个epoch耗时约13分40秒，GPU利用率在55%上下波动，CPU利用率约为110%，内存占用约2G，GPU显存占用约5.8G。
	最后，我在TDANet模型上将batch size增大到16，结果为：单个epoch耗时约5分40秒，GPU利用率保持在95%左右，CPU利用率约为130%，内存占用约2.5G，GPU显存占用约21.2G。
	我想了解这是否是TDANetBest正常的训练表现？如果不是，我可以采取哪些措施来提高TDANetBest的训练效率。
测试环境：
	显卡：NVidia 4090
	操作系统：Ubuntu 22.04.3 LTS
	Cuda version：12.1
	python version：3.10.8
	pytorch：2.1.2+cu121
模型参数配置：
	    out_channels: 128
	    in_channels: 512
	    num_blocks: 16
	    upsampling_depth: 5
	    enc_kernel_size: 4 # ms
	    num_sources: 2
其它参数配置：
          训练精度：混合精度
	  log_freq: 50
	  data_name: LRS2DataModule
	  data_config:
	    train_dir: DataPreProcess/LRS2mix/tr
	    valid_dir: DataPreProcess/LRS2mix/cv
	    test_dir: DataPreProcess/LRS2mix/tt
	    n_src: 2
	    sample_rate: 16000
	    segment: 2.0
	    normalize_audio: false
	    batch_size: 4
	    num_workers: 4
	    pin_memory: true
	    persistent_workers: false
感谢你的帮助，期待你的建议和回复。

**Comment by JusperLee** [comment]
我看超参数没有什么问题。训练精度：混合精度，这个可能会存在问题。

另外磁盘io也可能是影响利用率的一个因素。
